src/morphodict/lexicon/test_util.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+
+from morphodict.lexicon.util import to_source_language_keyword
+
+
+@pytest.mark.parametrize(
+    ("input", "expected"),
+    [
+        ("hello", "hello"),
+        ("hellø", "hello"),
+        ("hełło", "hello"),
+        ("hêllò", "hello"),
+        ("hěllo", "hello"),
+        (
+            "h̷̨̨̢̝̺̥͕̦̘̺̥̖̺̐ͅe̶̢̛͙̱͔̗͓͓̊̇̓̍̈́͐͋̍̐͑͌̌̿͘͜͠͝͝ļ̴̨̧͔̭̲̲͎̘̱̬̼͉̜̱̤͎̹̞̝̥̲̙̙̮̪̖̽̓̌͊̊̍̆́̎̑͌̇̓̈́́͊͌̾̆̿͊̎͌̈̕̕͜͠ͅl̸̨̧͓̭̻̝͖͕̪͖̩̯̞͙͖͙̦͈̘̥̱̩̫̬̎̈́͗̓̑̎̾͋̍̏͒̽̃̕͠ǒ̵͔̯̜̣̘̹̑͜͝",
+            "hello",
+        ),
```

Comment:
I would honestly want to see genuine examples from the languages we want to use _in addition_ to Unicode gore. Possible Unicode gore:

 - Greek casing: e.g., what should `to_source_language_keyword("σκύλος")` return?  Note that both `σ` and `ς` are lowercase Greek sigma (one is the final form). `.lower()` is context-sensitive, and not safe to apply on a per-character basis — you need to apply it on a full string.

There's more things you can bug me about, like Turkish, katakana fullwidth forms, etc. NFKD might be more appropriate for an "internal" representation.

Reply:
I did not realize that `lower()` isn’t safe on a character-by-character basis! The above code does actually run `lower()` on the whole string at the end, but I have moved it earlier in the function for clarity.

Reply:
[Lushootseed](https://en.wikipedia.org/wiki/Lushootseed) and/or [Nuuchahnulth](https://github.com/dwhieb/Nuuchahnulth#sounds-of-nuuchahnulth) would be good test languages, because they use COMBINING COMMA ABOVE (U+313), which inevitably causes problems.

Reply:
💯  I'VE HAD MANY PROBLEMS WITH U+0313. FONT SUPPORT IS VERY BAD :(

Reply:
I was very surprised that [the CBC used it successfully](https://www.cbc.ca/news/canada/british-columbia/heiltsuk-nation-indigenous-name-bc-government-identification-1.6093186) in a news story.

Reply:
How awful would it have been if they reported on that story and didn't manage to display her name right... 😬